### ISSUE TITLE
Suggestion: use OS defaults for calendar widget

### ISSUE THREAD

**Issue by mollieru** [18F/govt-wide-patternlibrary#87]
From @andrewmaier #38:

It's also really important to me that this be implement this in a modular/"progressively enhanced" way.  FilamentGroup has a number of plugins that they use to enhance forms and I've reliably customized them on client-facing work. See http://www.filamentgroup.com/code/ (esp. the code links under the "Accessible form elements" heading) Whenever possible, we should use HTML5 elements, such as `<input type="date">`, and *leave the default styling.* On an iPhone, for example, it presents the user with a really nice, touch friendly input:

![date](https://cloud.githubusercontent.com/assets/2970/8291668/ff070ed4-18f8-11e5-80ba-99b15206bcbe.png)

**Comment by maya** [comment]
Yes, we are planning on building in this way. Thanks for your input @andrewmaier!

**Comment by andrewmaier** [comment]
@maya Can you shed some light on how you're building this? Are you using `input type="date"` and then only showing a popup (custom, js-powered) calendar if the user agent _doesn't_ support touch events? I wouldn't recommend that approach because it rests on the assumption that mobile devices are the only devices with touch. (Remember that many Windows laptops support touch.)

On the other hand, styling native `input type=date` appears to be, well, not fun:
http://codepen.io/tgrant54/pen/LFblv

Curious what we're thinking.

**Comment by maya** [comment]
@andrewmaier we haven't started building that component yet. What approach do you recommend?

HTML5 with JS fallback? Any examples of folks doing it right / that we can adapt?

**Comment by andrewmaier** [comment]
I would just use the new (HTML) default. Maybe make the border match the other inputs. If the user agent doesn't support the new input type, try and use JS to mask the input so that it guides users to giving us the correct information. http://igorescobar.github.io/jQuery-Mask-Plugin/ (So long as there are instructions and example formats around that have `aria-describedby` attributes.)

**Comment by maya** [comment]
From @andrewmaier... 

TL;DR It’s a complex interaction and we don’t know as much as the operating system / user agent

**Comment by mollieru** [comment]
@andrewmaier Makes really good sense for mobile/tablet...how does this work on desktop?

**Comment by andrewmaier** [comment]
@mollieru The new normal is that there's "no such thing" as desktop.

Windows Surface, for example, is making very large touchscreen devices which means we may need to design for fat fingers (a large calendar) in the desktop world. That wouldn't fit on a small display, though, like a touchscreen phone. The easiest thing to do is to let the browser/OS determine what's the best way to handle the interaction.

**Comment by maya** [comment]
@mollieru This solution would mean we use browser defaults, so whatever Chrome, Safari, FF, IE use by default is how it would look and work.

**Comment by mollieru** [comment]
Totally into exploring it. I think it's relatively new territory for a bunch of folks so we'd need good grounding language and examples.  @andrewmaier  or @maya can you share some screenshots of what some of the browser defaults look like? 

Relatedly, is it safe to assume that Chrome, Safari, etc. are always going to be designing to the 508 standards we're after?  That'd be one of the only arguments to design our own if we're otherwise into this direction.

**Comment by mollieru** [comment]
Also, re-sharing a helpful link from gov.uk that @carodew snagged, re designing for a range of date-types:
https://www.gov.uk/service-manual/user-centred-design/resources/patterns/dates.html

**Comment by carodew** [comment]
I've been holding off on weighing in on the issue because I'm of two minds. I definitely understand and support the idea of not overriding the browser defaults – particularly on mobile. I'm not an expert on 508 but I want to support it as much as possible, so in the absence of strong evidence otherwise, I would tend to err on the side of 508 and accessibility over fancy inputs. 

I was thinking about this issue last night and was going to post the above link to GDS (so I'm glad @mollieru posted it!) With that in mind, I do think there are reasons to offer date entry methods that don't rely on a calendar picker (based on their arguments and examples). 

I also think that in some cases, the default calendar picker (which I believe only shows one month at a time, correct me if I'm wrong) is not the best solution. For example, when I'm booking flights and I don't have a specific weekend in mind, I often like to see 2-3 months at a time. So there may be examples in which a custom calendar picker would offer a solution otherwise unavailable – but is it worth the cost/risk of overriding the browser/OS defaults in cases where the calendar picker wouldn't be appropriate? I do agree with GDS that a JS calendar picker should never be the only option for input. 

I also wonder if people in government will just build their own calendar pickers anyway if we don't offer anything, even if we explain our reasons. I wonder if we should aim to offer a few different options (similar to GDS) and explain the pros/cons to each? I don't have the answers to any of these questions. 

I'd love to see links to any existing research on the issue and if I have a chance to look into it today, I'll share them here.

**Comment by juliaelman** [comment]
This has been resolved and has been added into the workflow.
